### PR TITLE
Replace .RawData with a dependency on ServiceControl.Monitoring.Data

### PR DESCRIPTION
**Pull request by Scooletz**
This PR removes `RawData` from the metrics package, introducing a dependency on the [ServiceControl.Monitoring.Data](https://github.com/Particular/ServiceControl.Monitoring.Data) and changes the library merging mechanism.

Following things are changed:
- `.RawData` is removed
- a dependency on `ServiceControl.Monitoring.Data` is taken. The dependency is il merged into the library so no package dependency is needed
- instead of IlMerge, ILRepack is used
- Metrics.NET are merged as they were before

**Warning**

Unfortunately I was not able to run tests in DEBUG, therefore I apply merging for DEBUG as well. This might be a stopper for merging this PR. Looking forward to comments on this. This was done in this commit https://github.com/Particular/NServiceBus.Metrics/pull/53/commits/659417b9700be13c815bb28bb063f181e276cfe3

### CODE REVIEW

**Comment by Scooletz** [comment]
It's WIP till https://github.com/Particular/ServiceControl.Monitoring.Data/issues/4 is done and a new package is released.

File: src/NServiceBus.Metrics/MetricsFeature.cs [hunk 1]
```diff
@@ -275,17 +277,32 @@ RawDataReporter CreateReporter(Action<Action<long, int>> setupProbe, string metr
...
+            {
+                var message = new OutgoingMessage(Guid.NewGuid().ToString(), reporterHeaders, body);
+
+                var operation = new TransportOperation(message, address);
+                try
+                {
+                    await dispatcher.Dispatch(new TransportOperations(operation), new TransportTransaction(), new ContextBag())
+                        .ConfigureAwait(false);
+                }
+                catch (Exception ex)
```
Comment (tmasternak) [inline src/NServiceBus.Metrics/MetricsFeature.cs]: Is it good enough that we swallow the exception and do not retry? Maybe we should extend [`RawDataReporter`](https://github.com/Particular/ServiceControl.Monitoring.Data/blob/master/src/ServiceControl.Monitoring.Data/RawDataReporter.cs#L90) with some retry policy? Either way sound like a separate PR.

Comment (Scooletz) [reply src/NServiceBus.Metrics/MetricsFeature.cs]: This is the same behavior as it was before. Also, it the ancient question: how to log/report when your logging/reporting fails. I see no other way than just reporting it for now.

**Comment by Scooletz** [comment]
The `ServiceControl.Monitoring.Data` has been bumped up. Which means it's SC compatible.

**Comment by Scooletz** [comment]
@tmasternak I find nothing against repacking in debug as well (pdbs are addressed as well).

**Comment by Scooletz** [comment]
@tmasternak I `git commit-tree` to not destroy the history of the PR and to put my changes on top of conflicting `develop` (your fix for tags). Please **do not merge it, but squash it`. We don't want to have history full of merges back and forth.

**Pull request merged**